task_02/nikolai_chernyshov/main.cpp

```diff
@@ -0,0 +1,175 @@
+#include <memory>
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+
+struct edge {
+    static int edge_id_max;
+    int edge_id;
```

Comment:
Это поле итак принадлежит структуре `Edge`, добавлять в название поля префикс `edge_` избыточно и только ухудшает читаемость кода.
```suggestion
    int id;
```

Reply:
Для лучшей выразительности языка можно `int` заменить на кастомный тип, например
```cpp
using VertexId = int;
```
В таком случае, если понадобится заменить `id` например на `std::string`, это нужно будет сделать только в одном месте.
```suggestion
    VertexId id;
```